src/main/java/org/spongepowered/common/data/manipulator/mutable/item/SpongePlaceableData.java

```diff
@@ -0,0 +1,91 @@
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.mutable.item;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.google.common.primitives.Booleans;
+import org.spongepowered.api.block.BlockType;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.item.ImmutablePlaceableData;
+import org.spongepowered.api.data.manipulator.mutable.item.PlaceableData;
+import org.spongepowered.api.data.value.mutable.SetValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+import org.spongepowered.common.data.manipulator.immutable.item.ImmutableSpongePlaceableData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractSingleData;
+import org.spongepowered.common.data.value.mutable.SpongeSetValue;
+
+import java.util.List;
+import java.util.Set;
+
+@NonnullByDefault
+public class SpongePlaceableData extends AbstractSingleData<Set<BlockType>, PlaceableData, ImmutablePlaceableData> implements PlaceableData {
+
+    public SpongePlaceableData() {
+        this(Sets.<BlockType>newHashSet());
+    }
+
+    public SpongePlaceableData(Set<BlockType> placeable) {
+        super(PlaceableData.class, placeable, Keys.PLACEABLE_BLOCKS);
+    }
+
+    @Override
+    public int compareTo(PlaceableData o) {
+        return Booleans.compare(o.placeable().containsAll(getValue()),
+                getValue().containsAll(o.placeable().get()));
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        List<String> placeableIds = Lists.newArrayListWithExpectedSize(getValue().size());
+        for (BlockType placeable : getValue()) {
+            placeableIds.add(placeable.getId());
+        }
+        return new MemoryDataContainer().set(Keys.PLACEABLE_BLOCKS.getQuery(), placeableIds);
+    }
+
+    @Override
+    public SetValue<BlockType> placeable() {
+        return new SpongeSetValue<BlockType>(Keys.PLACEABLE_BLOCKS, Sets.newHashSet(getValue()));
+    }
+
+    @Override
+    public PlaceableData copy() {
+        return new SpongePlaceableData(Sets.newHashSet(getValue()));
+    }
+
+    @Override
+    public ImmutablePlaceableData asImmutable() {
+        return new ImmutableSpongePlaceableData(getValue());
```

Comment:
You should store any `ImmutableValue` or `ImmutableDataManipulator` instances in a field, and obtain them using `ImmutableDataCachinUtil`.